File: python/types/integers/integer_constructor.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

i = int(0)
print(f"i = int(0) # {i} ({type(i)})")

i = int(5)
print(f"i = int(5) # {i} ({type(i)})")

i = int(-5)
print(f"i = int(-5) # {i} ({type(i)})")

i = int(5.2)
print(f"i = int(5.2) # {i} ({type(i)})")

i = int("0")
print(f"i = int(\"0\") # {i} ({type(i)})")

i = int("5")
print(f"i = int(\"5\") # {i} ({type(i)})")

i = int("-5")
print(f"i = int(\"-5\") # {i} ({type(i)})")

# i = int("5.2") # ValueError: invalid literal for int() with base 10: '5.2'
# print(f"i = int(\"-5.2\") # {i} ({type(i)})")

File: ruby/types/integers/integer_constructor.rb
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md

i = Integer(0)
puts "i = Integer(0) # #{i} (#{i.class})"

i = Integer(5)
puts "i = Integer(5) # #{i} (#{i.class})"

i = Integer(-5)
puts "i = Integer(-5) # #{i} (#{i.class})"

i = Integer(5.2)
puts "i = Integer(5.2) # #{i} (#{i.class})"

i = Integer("0")
puts "i = Integer(\"0\") # #{i} (#{i.class})"

i = Integer("5")
puts "i = Integer(\"5\") # #{i} (#{i.class})"

i = Integer("-5")
puts "i = Integer(\"-5\") # #{i} (#{i.class})"

# i = Integer("5.2") # invalid value for Integer(): "5.2" (ArgumentError)
# puts "i = Integer(\"5.2\") # #{i} (#{i.class})"
